### PR TITLE
Issue 144 implement cd() only

### PR DESCRIPTION
**Pull request by ntwrick**
Want to get `cd()` into the coming release.

The remaining OS function set is pretty big, so we live to fight another day...

### CODE REVIEW

File: evaluator/functions.go [hunk 1]
```diff
@@ -411,6 +416,28 @@ func pwdFn(args ...object.Object) object.Object {
...
+func cdFn(args ...object.Object) object.Object {
+	// Default cd $HOME
+	path := os.Getenv("HOME")
+	if len(args) == 1 {
+		// arg: path
+		pathStr := args[0].(*object.String)
+		rawPath := pathStr.Value
+		// expand ~/ path prefix
+		if strings.HasPrefix(rawPath, "~/") {
+			path = strings.Replace(rawPath, "~/", path+"/", 1)
```
Comment (odino) [inline evaluator/functions.go]: I think this might not work under windows :)

Comment (ntwrick) [reply evaluator/functions.go]: Good call. I'll make it portable.

**Comment by odino** [comment]
Also, you could possibly use [HomeDir](https://golang.org/src/os/user/user.go?s=892:1756#L19).

**Comment by odino** [comment]
@ntwrick BTW [I just released 1.1.0](https://github.com/abs-lang/abs/releases/tag/1.1.0) -- loads of stuff was pending and it might be a few more days until I get the time to draft a release, so thought it would be better to do it today and move all the remaining issues to [1.2](https://github.com/abs-lang/abs/milestone/9).

**Comment by ntwrick** [comment]
Thanks for the `"os/user" HomeDir` tip. This works well in both linux and windows.

The new and improved `cd()` command now always returns a `path` string instead of error-ing out the entire program `eval()`. This is nice for working with `relative paths` such as `cd("..")` or `expanded paths` such as `cd("~/my/path")` since you get back the full `absolute path` string. 

This approach also allows the user to control how the error is handled via `path.ok` and to do  tests like `path = cd("path/to/nowhere"); if path.ok {...}` as well as compound tests such as``dirs = cd() && `ls`.lines(); len(dirs)``.

And as a bonus, `cd("~/my/path")` expands to `"/home/user/my/path"` in linux while windows expands to `"C:\Users\user"`. In addition, something linuxy like `cd("~/downloads")` in windows returns `C:\Users\user\Downloads`.

**Comment by ntwrick** [comment]
Ping... @odino ... is this PR ok to merge? It has been tested in both linux and win10.

**Comment by odino** [comment]
Hey @ntwrick this looks good! When I see the files changed though I see some conflicts with `1.2.x` as well as a truckload of commits :) maybe a funny issue with github. Do you mind rebasing / re-sending the PR?

**Comment by ntwrick** [comment]
Closing this PR in favor of #165.

**Pull request closed without merging**